peps/pep-0751.rst

```diff
@@ -0,0 +1,901 @@
+  required for per-file locking as the ``package.*.lock`` entries imply mutual
+  exclusivity)
+- Aids in auditing by knowing that there are multiple entries for the same
+  package that may need to be considered
+
+
+``package.description``
+'''''''''''''''''''''''
+
+- Optional
+- String
+- The package's ``Summary`` from its `core metadata`_
+- Useful to help understand why a package was included in the file based on its
+  purpose
+
+
+``package.simple-repo-package-url``
+'''''''''''''''''''''''''''''''''''
+
+- Optional (although mutually exclusive with
+  ``package.files.simple-repo-package-url``)
+- String
+- Stores the `project detail`_ URL from the `Simple Repository API`_
+- Useful for generating Packaging URLs (aka *PURLs*)
+- When possible, lockers SHOULD include this or
+  ``package.files.simple-repo-package-url`` to assist with generating
+  `software bill of materials`_ (aka SBOMs)
+
+
+``package.marker``
+''''''''''''''''''
+
+- Optional
+- String
+- The `environment markers`_ expression which specifies whether this package and
+  version applies to the environment
+- Only applicable via ``[package-lock]`` and the package locking scenario
+- The lack of this key means this package and version is required to be
+  installed
+
+
+``package.requires-python``
+'''''''''''''''''''''''''''
+
+- Optional
+- String
+- Holds the `version specifiers`_ for Python version compatibility for the
+  package and version
+- Useful for documenting why this package and version was included in the file
+- Also helps document why the version restriction in
+  ``package-lock.requires-python`` was chosen
+- It should not provide useful information for installers as it would be
+  captured by ``package-lock.requires-python`` and isn't relevant when
+  ``[[file-lock]]`` is used
+
+
+``package.dependents``
+''''''''''''''''''''''
+
+- Optional
+- Array of strings
+- A record of the packages that depend on this package and version
+- Useful for analyzing why a package happens to be listed in the file
+  for auditing purposes
+- This does not provide information which influences installers
+
+
+``package.dependencies``
+''''''''''''''''''''''''
+
+- Optional
+- Array of strings
+- A record the dependencies of the package and version
+- Useful in analyzing why a package happens to be listed in the file
+  for auditing purposes
+- This does not provide information which influences the installer as
+  ``[[file-lock]]`` specifies the exact files to use and ``[package-lock]``
+  applicability is determined by ``package.marker``
+
+
+``package.direct``
+''''''''''''''''''
+
+- Optional
+- Boolean
+- Represents whether the installation is via a `direct URL reference`_
+
+
+``[[package.files]]``
+'''''''''''''''''''''
+
+- Must be specified if ``[package.vcs]`` is not
+- Array of tables
+- Tables can be written inline
+- Represents the files to potentially install for the package and version
```

Comment:
Should we specify a recommended order for the entries?

Reply:
Yes. Added a note under `package.file.name`.